codebuild/spec/buildspec_ktls.yml

```diff
@@ -11,26 +11,37 @@
 # "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
 # implied. See the License for the specific language governing permissions and
 # limitations under the License.
+# This is designed to work with CodeBuild's reserved instances fleet and curated Ec2 AMI for AL2023.
 version: 0.2
+env:
+  shell: bash
+  variables:
+    NIX_CACHE_BUCKET: "s3://s2n-tls-nixcachebucket-x86-64?region=us-west-2"
+    S2N_KTLS_TESTING_EXPECTED: 1
 phases:
   install:
     commands:
-      - apt update
-      - apt upgrade -y
-      - apt install -y qemu qemu-system-x86 wget cloud-image-utils
+      - yum update -y; yum upgrade -y
   pre_build:
     commands:
-      - aws s3 --quiet sync s3://s2n-ktls-testing ./qemu
-      - cd qemu; bash ./run.sh; cd ..
-      - rsync -avz --exclude=qemu --exclude=tests/fuzz -e 'ssh -p 2222' . codebuild@localhost:/home/codebuild/s2n-tls
+      - id; groupadd nixbld||true
+      - useradd -m -g nixbld -G nixbld nix || true
+      - |
+        echo "Working around the faulty yaml parser..."
+        echo 'nix ALL=NOPASSWD: ALL' > /etc/sudoers.d/nix
+      # (Re)Install nix
+      - sh <(curl -L https://nixos.org/nix/install) --no-daemon
+      # Make sure nix exists in the PATH
+      - export PATH=$HOME/.nix-profile/bin:$PATH
+      # Turn on flakes
+      - mkdir -p ~/.config/nix; echo "experimental-features = nix-command flakes" >> ~/.config/nix/nix.conf
+      # Populate the store from the nix cache
+      - nix copy --from $NIX_CACHE_BUCKET --all  --no-check-sigs
```

Comment:
Do we need to do all of this whenever we want to use nix on codebuild? Should there be a reusable script for this?

Reply:
No, this was my MVP...This gets nix working on an AL2023 instance; which could be dropped if we moved to a NixOS AMI.